test-tools/aot-analyzer/src/main.cc

```diff
@@ -0,0 +1,656 @@
+    for (uint32_t index = 0; index < import_memory_count; index++) {
+        AOTImportMemory memory = import_memories[index];
+        printf("    -[%u] num_bytes_per_page:%5u    init_page_count:%5u    "
+               "max_page_count:%5u    module_name: %s    memory_name: %s \n",
+               index, memory.num_bytes_per_page, memory.mem_init_page_count,
+               memory.mem_max_page_count, memory.module_name,
+               memory.memory_name);
+    }
+    printf("\n");
+
+    // import tables
+    printf("  -import_tables[%u] \n", import_table_count);
+    for (uint32_t index = 0; index < import_table_count; index++) {
+        AOTImportTable table = import_tables[index];
+        printf("    -[%u] ", index);
+        printf("elem_type: ");
+#if WASM_ENABLE_GC != 0
+        wasm_dump_value_type(table.elem_type, table.elem_ref_type);
+#else
+        dump_value_type(table.elem_type);
+#endif
+        printf("    init_size:%5u    max_size:%5u    "
+               "module_name: %s    table_name: %s \n",
+               table.table_init_size, table.table_max_size, table.module_name,
+               table.table_name);
+    }
+    printf("\n");
+
+    // import globals
+    printf("  -import_globals[%u] \n", import_global_count);
+    for (uint32_t index = 0; index < import_global_count; index++) {
+        AOTImportGlobal global = import_globals[index];
+        printf("    -[%u] ", index);
+        printf("type: ");
+        dump_value_type(global.type);
+        printf("    module_name: %s    global_name: %s \n", global.module_name,
+               global.global_name);
+    }
+    printf("\n");
+
+    // import functions
+    printf("  -import_functions[%u] \n", import_func_count);
+    for (uint32_t index = 0; index < import_func_count; index++) {
+        AOTImportFunc func = import_funcs[index];
+        printf("    -[%u] module_name: %s    function_name: %s \n", index,
+               func.module_name, func.func_name);
+    }
+    printf("\n\n");
+
+    // Functions
+    const uint32_t func_count = ((AOTModule *)(aot->GetModule()))->func_count;
+    const uint32_t code_size = ((AOTModule *)(aot->GetModule()))->code_size;
+    struct func_info *sorted_func_ptrs = NULL;
+    sorted_func_ptrs = sort_func_ptrs(((AOTModule *)(aot->GetModule())));
+
+    if (sorted_func_ptrs) {
+        printf("Function[%u] \n", func_count);
+        for (uint32_t index = 0; index < func_count; index++) {
+            const uint32_t func_size =
+                index + 1 < func_count
+                    ? (uintptr_t)(sorted_func_ptrs[index + 1].ptr)
+                          - (uintptr_t)(sorted_func_ptrs[index].ptr)
+                    : code_size
+                          + (uintptr_t)(((AOTModule *)(aot->GetModule()))->code)
+                          - (uintptr_t)(sorted_func_ptrs[index].ptr);
+            printf("  -[%u] code_size= %u Bytes \n", index, func_size);
+        }
+        printf("\n\n");
```

Comment:
Should `wasm_runtime_free(sorted_func_ptrs)` when it isn't used.

Reply:
done